.changelog/18745.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+server: Add `go-netaddrs` support to `retry_join`
```

Comment:
This impacts both client and server. The bare `retry_join` config is [deprecated](https://developer.hashicorp.com/nomad/docs/configuration/server#deprecated-parameters) so let's refer specifically to the one we most care about:

```suggestion
config: Add `go-netaddrs` support to `server_join.retry_join`
```